crates/ruff/src/rules/ruff/rules/mutable_defaults_in_class_fields.rs

```diff
@@ -186,23 +221,41 @@ pub fn function_call_in_dataclass_defaults(checker: &mut Checker, body: &[Stmt])
                 continue;
             }
             if let ExprKind::Call { func, .. } = &expr.node {
-                if !is_immutable_func(&checker.ctx, func, &extend_immutable_calls)
-                    && !is_allowed_dataclass_function(&checker.ctx, func)
+                if !(is_immutable_func(&checker.ctx, func, &extend_immutable_calls)
+                    || is_dataclass && is_allowed_dataclass_function(&checker.ctx, func))
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        FunctionCallInDataclassDefaultArgument {
-                            name: compose_call_path(func),
-                        },
-                        expr.range(),
-                    ));
+                    let diagnostic: Diagnostic = if emit_dataclass_error {
+                        Diagnostic::new(
+                            FunctionCallInDataclassDefaultArgument {
+                                name: compose_call_path(func),
+                            },
+                            expr.range(),
+                        )
+                    } else {
+                        Diagnostic::new(
+                            FunctionCallInClassDefaultArgument {
+                                name: compose_call_path(func),
+                            },
+                            expr.range(),
+                        )
+                    };
+                    checker.diagnostics.push(diagnostic);
                 }
             }
         }
     }
 }
 
-/// RUF008
-pub fn mutable_dataclass_default(checker: &mut Checker, body: &[Stmt]) {
+/// RUF008/RUF010
+pub fn mutable_class_default(checker: &mut Checker, emit_dataclass_error: bool, body: &[Stmt]) {
```

Comment:
Nit: It can be difficult to understand the semantic of `true` in a call of `mutable_class_default(checker, true, body)` without looking at the method signature. 

Introducing an enum can improve readability:

```rust
#[derive(Copy, Clone, Debug)]
enum ClassKind {
	Class,
	Dataclass
}
```

It has the added benefit that you can implement methods on the kind